mesonbuild/mintro.py

```diff
@@ -53,7 +53,7 @@ def __init__(self,
 def get_meson_introspection_types(coredata: T.Optional[cdata.CoreData] = None,
                                   builddata: T.Optional[build.Build] = None,
                                   backend: T.Optional[backends.Backend] = None,
-                                  sourcedir: T.Optional[str] = None) -> 'T.OrderedDict[str, IntroCommand]':
+                                  sourcedir: T.Optional[str] = None) -> 'collections.OrderedDict[str, IntroCommand]':
```

Comment:
```suggestion
                                  sourcedir: T.Optional[str] = None) -> 'T.Mapping[str, IntroCommand]':
```
Might be what you want.

Reply:
It's not as exact, but it is correct and exists in all versions of python we try to support.

Reply:
But would that be any different from using T.Dict (as it was in the beginning)? As in using a different type, which has the same interface as our target type.

Reply:
Because Dict is a type of Mapping, and OrderedDict is a type of Mapping, but OrderedDict is not a type of Dict (or vice versa)

Reply:
Fixed